sycl/plugins/level_zero/pi_level0.cpp

```diff
@@ -608,6 +608,12 @@ pi_result piDevicesGet(pi_platform Platform, pi_device_type DeviceType,
   if (NumDevices)
     *NumDevices = ZeDeviceCount;
 
+  if (Devices == nullptr) {
+    assert(NumEntries == 0 &&
+           "NumEntries should be zero when querying the number of devices");
+    return PI_SUCCESS;
+  }
+
```

Comment:
Should we move this before `zeDeviceGet` call?

Reply:
This code intended to early exit when numEntries == 0, which is the case when the caller just queries the number of devices.
We can't move this early exit before the zeDevicesGet() which is responsible for getting the number of devices.

Reply:
I see. It's not obvious and I was confused because this early exit depends on `Devices` and `NumEntries` variables and doesn't write to `NumDevices` inside the if-branch.